Refactor documentation and add project overview sections (#19)
eProsima/DDS-Recorder: docs/rst/developer_manual/installation/sources/windows.rst

- ``cpp_utils``, an eProsima utils library for C++.

> - ``ddsrouter``, an eProsima end-user application that enables the connection of distributed DDS networks.

Reviewer: ```suggestion
- ``ddspipe``, an eProsima internal library that enables the communication of DDS interfaces.
```

eProsima dependencies
^^^^^^^^^^^^^^^^^^^^^

> If it already exists in the system an installation of *Fast DDS* and *DDS Router* libraries, just source this libraries when building the |eddsrecord| application by using the command:

Reviewer: ```suggestion
If it already exists in the system an installation of *Fast DDS* and *DDS Pipe* libraries, just source this libraries when building the |eddsrecord| application by using the command:
```

.. code-block:: bash

    source <fastdds-installation-path>/install/setup.bash

>     source <ddsrouter-installation-path>/install/setup.bash

Reviewer: ```suggestion
    source <ddspipe-installation-path>/install/setup.bash
```

Global installation
-------------------

> To install |eddsrecord| system-wide instead of locally, remove all the flags that appear in the configuration steps of :code:`Fast-CDR`, :code:`Fast-DDS`, :code:`Dev-Utils`, :code:`DDS-Pipe`, and :code:`DDS-Recorder`

Reviewer: ```suggestion
To install |eddsrecord| system-wide instead of locally, remove all the flags that appear in the configuration steps of :code:`Fast-CDR`, :code:`Fast-DDS`, :code:`Dev-Utils`, :code:`DDS-Pipe`, and :code:`DDS-Record`
```

Run an application
==================

> If the |eddsrecord| was compiled using colcon, when running an instance of a |ddsrecorder|, the colcon overlay built in the dedicated :code:`DDS-Recorder` directory must be sourced.

Reviewer: ```suggestion
If the |eddsrecord| was compiled using colcon, when running an instance of a |ddsrecorder|, the colcon overlay built in the dedicated :code:`DDS-Record` directory must be sourced.
```
